### DISCUSSION TITLE
Add "operating state" to Humidifier entity

### DISCUSSION THREAD

**Discussion by EiNSTeiN-** [home-assistant/architecture#731] [Entity Models] [answered]
Many humidifiers have the ability to report on their current state, but `HumidifierEntity` has no way to make use of it. I did a quick survey of implementations and I found:
* homekit has [`CurrentHumidifierDehumidifierState`](https://github.com/Colorado4Wheeler/HomeKit-Bridge/wiki/HomeKit-Model-Reference#currenthumidifierdehumidifierstate)
* ecobee has [`equipmentStatus`](https://www.ecobee.com/home/developer/api/documentation/v1/objects/Thermostat.shtml)
* Z-Wave humidity control command class has an [`Operating state`](https://github.com/home-assistant-libs/zwave-js-server-python/blob/f854a2759fec88548fe6b65ae261243898a66539/zwave_js_server/const/command_class/humidity_control.py#L27-L33). 

Devices that actively humidify or de-humidify (eg. via steam or a fan) would benefit from this, since turning "on" the device does not mean it is currently operating. It is the case for my thermostat for instance, it will switch between `Humidifying` and `Idle` depending on the current humidity level.

The `GenericHygrostat` is a good candidate for this proposal as well. Its state depends on a combination of the current and desired humidity, the mode it is set to and the minimum cycle duration that is configured. Exposing the internal state to users would bring an additional level of clarity on what part of the cycle the entity is currently in.

The Humidifier card would then be able to display the "operating state" to give an additional level of feedback to users, like this

| ![lovelace_humidifier_card](https://user-images.githubusercontent.com/2053039/152693584-c39c49e1-354e-43d9-8eea-e14f5a52c72e.png) | ![lovelace_humidifier_card (copy)](https://user-images.githubusercontent.com/2053039/152693583-df66d9ff-bdaf-4220-91da-3173f46c1e3d.png) |
|---|---|

This is my proposed implementation which can serve as a basis for this discussion: https://github.com/home-assistant/core/pull/65855

**Comment by EiNSTeiN-** [comment]
@Shulyaka since you mostly wrote the humidifier integration, would you have an opinion on this proposal?

**Comment by Shulyaka** [comment]
I personally love the idea.
We see that some users don't fully understand the difference between being on and being active.
I don't know if it is in line with the modern architecture trends to split everything into separate entities, but this property is better to be included as an attribute rather than a separate sensor, since it is also embedded in other implementations, and we also don't have an operation state sensor device type, so my vote is for your solution.

**Comment by frenck** [comment, marked as answer]
We have discussed this proposal today, and agreed this can be added to the humidity entity component.

We do want to suggest avoiding the use of "operating state" as it closely resembles "state", which might be confusing. We recommend using "action" instead (which is closer to the "hvac_action" the climate model uses).

../Frenck

**Reply by Shulyaka** [reply]
Thanks! What about the `current_status` or `status` with values `idle` or `active`?

**Reply by frenck** [reply]
We like to stick with the terms used in climate as much as possible.

**Reply by Shulyaka** [reply]
Okay!

**Reply by Shulyaka** [reply]
What about the values?

Options:

1. "idle", "active"
2. "idle", "humidifying", "dehumidifying"
3. "idle", "humidifying", "drying"
4. "off", "idle", "active"
5. "off", "idle", "humidifying", "dehumidifying"
6. "off", "idle", "humidifying", "drying"

From my perspective, "off" is redundant. Also it could be set to None when state is "off".
Also "humidifying" and "drying"/"dehumidifying" cannot be used in the same entity.

**Reply by gjohansson-ST** [reply]
I believe option 6 is the right one which also aligns well with `climate`

**Comment by frenck** [comment]
This feature has been added to Core. Thanks all!

../Frenck
